src/vector/_methods.py

```diff
@@ -2966,32 +2966,97 @@ def to_Vector3D(
         theta: float | None = None,
         eta: float | None = None,
     ) -> VectorProtocolSpatial:
+        """
+        Converts a 2D vector to 3D vector.
+
+        The scalar longitudinal coordinate is broadcasted for NumPy and Awkward
+        vectors. Only a single longitudinal coordinate should be provided. Generic
+        coordinate counterparts should be provided for the momentum coordinates.
+
+        Examples:
+            >>> import vector
+            >>> vec = vector.VectorObject2D(x=1, y=2)
+            >>> vec.to_Vector3D(z=1)
+            VectorObject3D(x=1, y=2, z=1)
+            >>> vec = vector.MomentumObject2D(px=1, py=2)
+            >>> vec.to_Vector3D(z=4)
+            MomentumObject3D(px=1, py=2, pz=4)
+        """
         if sum(x is not None for x in (z, theta, eta)) > 1:
-            raise TypeError("Only one non-None parameter allowed")
+            raise TypeError("Only one non-None longitudinal coordinate allowed")
```

Comment:
The above is helpful; maybe this is even more helpful (and it sets the tone for what the other `to_Vector*D` should do).

```suggestion
            raise TypeError("At most one longitudinal coordinate (`z`, `theta`, or `eta`) may be assigned (non-None)")
```

By the way, these `to_Vector*D` functions maintain momentum-ness, right? For example, is it the case that `Momentum2D.to_Vector3D` returns a `Momentum3D`?

If so, then a user might justifiably expect to pass `pz` instead or `z`, but only in the momentum case. Just something to consider, at this point.

When this gets extended to `to_Vector4D`, the time component has a lot of momentum synonyms. The geometric names for the two possible temporal coordinates are `t` and `tau`, but the momentum names are `energy` and `mass`, `E` and `M`, and (I think) `e` and `m`. There was a discussion once about including the lowercase as well as the uppercase.

Reply:
Yes, they do maintain momentum-ness. Right now the generic coordinates are automatically converted to momentum coordinates, but yes, we can explicitly add momentum coordinates as arguments in the future -

```py
In [1]: import vector

In [2]: vector.obj(px=1, py=2).to_Vector4D(z=3, t=4)
Out[2]: MomentumObject4D(px=1, py=2, pz=3, E=4)

In [3]: vector.obj(px=1, py=2).to_Vector4D(eta=3, tau=4)
Out[3]: MomentumObject4D(px=1, py=2, eta=3, mass=4)
```